### ISSUE TITLE
Problem downloading anything with plugin manager

### ISSUE THREAD

**Issue by embeh** [bruderstein/nppPluginManager#86]
### Description of the Issue
Plugin manager refuses to download plugin list or updates.

### Steps to Reproduce the Issue
0. Manually install Plugin Manager by copying PluginManager.dll to C:\Program Files (x86)\Notepad++\plugins and gpup.exe to C:\Program Files (x86)\Notepad++\updater

1.  Start n++

2. Plugins > Plugin Manager > Show Plugin Manager

3. Plugin Manager reports 'Downloading plugin list...' for about a minute and then reports back with an error message:

```
---------------------------
Download Error
---------------------------
There was an error downloading the plugin list. Please check your internet connection, and your proxy settings in Internet Explorer, Edge or Chrome
---------------------------
OK   
---------------------------
```

4. In the Plugin Manager settings, enable 'Use development plugin list' and restart n++ ==> a long list of plugins is now available.

5. Under the 'Installed' tab, check 'Plugin Manager' and click on 'Reinstall' ==> some downloading from github seems to be occurring ==> an error message is presented:

```
---------------------------
Installation Error
---------------------------
Installation of Plugin Manager failed.
---------------------------
OK   
---------------------------
```

6. Turn off 'Use development plugin list' --> Plugin manager displays a 'Refreshing plugin list..' information for about a minute and reports the same error from step number 3

7. Enable 'Force HTTP instead of HTTPS for downloading Plugin List' and restart n++, then repeat steps 2 and 3 ==> same result as above in step 3

8. Doing the everything above (steps 1-7) with 'Run as administrator' produces the same results. 

### Expected Behavior
Should download plugin list normally, allow to install new plugins and update existing ones.

### Actual Behavior
Error message (see above)

### Debug Information
Notepad++ v7.5.4   (32-bit)
Build time : Jan  1 2018 - 01:47:45
Path : C:\Program Files (x86)\Notepad++\notepad++.exe
Admin mode : OFF
Local Conf mode : OFF
OS : Windows 7 (64-bit)
Plugins : AnalysePlugin.dll ComparePlugin.dll DSpellCheck.dll IndentByFold.dll mimeTools.dll NppCCompletionPlugin.dll NppConverter.dll nppcrypt.dll NppExport.dll NppFTP.dll NppTextFX.dll PluginManager.dll PythonScript.dll regrexplace.dll SpellChecker.dll XMLTools.dll 

nppPluginManager version 1.4.9.0

Maybe the problem is related to downloads from sourceforge.net not working, whereas downloads from github seem to be ok?

Other internet connectivity is fine, including downloads from sourceforge using Firefox.

Is there any way to get more log output from Plugin Manager?

**Comment by chcg** [comment]
@embeh In general everything seems to be still fine for me. Also sourceforge downloads and installations are still ok. Tested with AnalysePlugin and AutoCodepage. Are you still facing this problems?

The normal plugin list is downloaded from https://nppxml.bruderste.in/pm/xml/plugins.zip.
Could you access that location without problems? Maybe the internet traffic there was to high or the server overloaded.

The additional locations for win32:

```
#define PLUGINS_MD5_URL     _T("https://nppxml.bruderste.in/pm/xml/plugins2.md5.txt")
#define PLUGINS_HTTP_MD5_URL     _T("http://nppxml.bruderste.in/pm/xml/plugins2.md5.txt")
#define PLUGINS_URL         _T("https://nppxml.bruderste.in/pm/xml/plugins.zip")
#define PLUGINS_HTTP_URL         _T("http://nppxml.bruderste.in/pm/xml/plugins.zip")

#define DEV_PLUGINS_MD5_URL     _T("https://nppxmldev.bruderste.in/pm/xml/plugins2.md5.txt")
#define DEV_PLUGINS_URL _T("https://nppxmldev.bruderste.in/pm/xml/plugins.zip")
```

**Comment by embeh** [comment]
Yes, that download for the normal plugin list works just fine in Firefox, on the same machine.
I have this problem now for an extended period of time, so it's not something that just came up in recent days, it's something more fundamental.
It's related to the newer plugin manager versions (those with the nexinto logo in the bottom left). Before that, things seemed to work ok.
FWIW, I just noticed that I have PluginManager.ini and other files in %APPDATA%\Notepad++\plugins\Config\PluginManager*  as well as in C:\Program Files (x86)\Notepad++\plugins\PluginManager.dll and C:\Program Files (x86)\Notepad++\updater\gpup.exe

**Comment by chcg** [comment]
@embeh  Maybe also an issue with the supported TLS version of schannel on WIN7, see https://blogs.msdn.microsoft.com/kaushal/2011/10/02/support-for-ssltls-protocols-on-windows/. Could you please check if the download is functional with IE. The PluginManager uses the same windows schannel component for encrypted transfers. Maybe in the meantime it is required to have TLS 1.1 and 1.2 enabled for access to some websites. Current firefox has its own SSL/TLS implementation, so this can't be taken as reference for schannel based connections.

**Comment by cpmcgrath** [comment]
Try going to the properties of each of the files for the plugin manager, If at the bottom there's a checkbox to unblock it, tick it and press okay.

**Comment by embeh** [comment]
@chcg - I am not really using IE a lot, but I have IE11 installed and that was fine to download from https://nppxml.bruderste.in/pm/xml/plugins.zip without errors.
How would I enable TLS 1.1 and 1.2?

@cpmcgrath - assuming you are referring to unblocking the files on the file system, I followed https://blogs.msdn.microsoft.com/delay/p/unblockingdownloadedfile/ and checked C:\Program Files (x86)\Notepad++\plugins\PluginManager.dll as well as all files in C:\Program Files (x86)\Notepad++\updater but none of them had the Unblock button. I also unblocked the original .zip file for the plugin manager and extracted its content again to the npp folders.

All without improvement.

**Comment by embeh** [comment]
FWIW, I uninstalled NPP, renamed the old directory in the programs folder and the npp user folder. Then I downloaded a fresh installer of NPP 7.5.4 32-bit and did a new install.
Next, I downloaded PluginManager_v1.4.9_UNI.zip and copied the two files in their respective npp folders - and guess what? **The same problem**!

Any download shows zero progress and eventually times out and I get

---------------------------
Installation Error
---------------------------
Installation of Plugin Manager failed.
---------------------------
OK   
---------------------------

Any suggestions what I could do to get plugin manager working again?!

**Comment by embeh** [comment]
Here's a screenshot of the security options in IE:
![ie_sec_options](https://user-images.githubusercontent.com/3431046/35437346-4d933c3c-0292-11e8-908d-fd805cd036e9.png)

**Comment by embeh** [comment]
OK...........

Here's the solution: @chcg was on the right track, it was a problem with Internet Explorer settings. But it was not related to SSL, TLS or such things.

It was *drumroll* the cookie settings! My IE is configured to ask (allow/block) whenever a site wants to set a cookie. And there is an option to 'remember' that. 

So, manually going to github and sourceforge once with IE and blocking cookies and remembering it has fixed this.

Bottom line: it's not great to use IE components for downloading stuff off the web, and it's not great to get stuck on an invisible popup note asking for cookie confirmation.... ;-)

**Comment by rsandri91** [comment]
The problem is that Internet Explorer thinks the web certificate for the website with he plugin list is untrusted so that affects any app running on Windows that looks to the IE settings for guidance on how to deal with the site.  I had every problem you listed here exactly and my problems were resolved by forcing HTTP instead of HTTPS.  If you try to browse to the URL using IE instead of Firefox you will see what I mean.

**Comment by rsandri91** [comment]
The problem is that Internet Explorer thinks the web certificate for the website with he plugin list are untrusted so that affects any app running on Windows that looks to the IE settings for guidance on how to deal with the site.  I had every problem you listed here exactly and my problems were resolved by forcing HTTP instead of HTTPS.  If you try to browse to the URL using IE instead of Firefox you will see what I mean.

**Comment by n0sr3v** [comment]
Guys thank you so much for your help. Especially the hint on the fact that plugin-manager is using IE components! My workplace IE settings seemed to be messed up. Was not able to access github.com (via IE) at all.

IE actually tells you what to to: "Make sure TLS and SSL protocols are enabled. Go to Tools > Internet Options > Advanced > Settings > Security"

Only TLS 1.0 and 1.1 have been enabled for me :tada: 
![internetexplorer_missing_settings](https://user-images.githubusercontent.com/22380455/42422283-121185de-82b1-11e8-83ac-d5903f30a953.PNG)

**Comment by vinczej** [comment]
I didn't think, that IE can affect the Notepad++  ( :O ), but indeed, in the settings of IE (TLS 1.2 enable) was the solution for my same problem! Thanks! :)

**Comment by vasmedv** [comment]
Thank you @n0sr3v, turning on TSL 1.1 and 1.2 worked for me.

**Comment by danielbighelini** [comment]
Thank you @n0sr3v, turning on TSL 1.1 and 1.2 worked for me.

**Comment by KimEbelACE** [comment]
I am running into a similar problem. I installed Plugin Manager manually by downloading the latest version and moving the gpup.exe and PluginManager.dll files to the plugins folder and got the error message "There was an error downloading the plugin list. Please check your internet connection, and your proxy settings in Internet Explorer, Edge or Chrome" when selecting "show plugin manager". 

I tried enabling the development plugin list and restarting notepad++ - no change. 
I went to my Internet Options settings (via IE) and added github.com and sourceforge.net to the "Always block" cookies list and restarted notepad++ - no change.
I tried enabling the "Force HTTP instead of HTTPS for downloading plugin list" and restarting notepad++ - no change. 
I checked my internet options settings, and "use TLS 1.0/1.1/1.2" are already enabled.

I'm not sure what other information may be needed for solving this problem, but any advice would be greatly appreciated!

**Comment by chcg** [comment]
@KimEbelACE Which version of N++? PM is just usable up to version 7.5.9. Newer N++ versions 7.6.x have the builtin PluginAdmin which should be used.
PluginManager dev list is not working because the certificate is not renewed yet. 
@bruderstein Could you check that?

**Comment by KimEbelACE** [comment]
@chcg I'm currently using N++ v7.5.9 (64-bit).  I did run the "Update N++" option while troubleshooting (I forgot to mention it in the post above) and it said "No update is available", so I wasn't aware there were newer versions.

**Comment by chcg** [comment]
@KimEbelACE N++ updates are not available for every release, just after some time of testing and if the version is considered to be stable an N++ update is available. In case of 7.6.x due to the PluginAdmin changes so far no version is considered as stable enough for a release.
